inotify_test.go

```diff
@@ -157,92 +156,105 @@ func TestInotifyCloseCreate(t *testing.T) {
 func TestInotifyStress(t *testing.T) {
 	testDir := tempMkdir(t)
 	defer os.RemoveAll(testDir)
-	testFile := filepath.Join(testDir, "testfile")
+	testFilePrefix := filepath.Join(testDir, "testfile")
 
 	w, err := NewWatcher()
 	if err != nil {
 		t.Fatalf("Failed to create watcher: %v", err)
 	}
 	defer w.Close()
 
-	killchan := make(chan struct{})
-	defer close(killchan)
-
 	err = w.Add(testDir)
 	if err != nil {
 		t.Fatalf("Failed to add testDir: %v", err)
 	}
 
-	proc, err := os.FindProcess(os.Getpid())
-	if err != nil {
-		t.Fatalf("Error finding process: %v", err)
-	}
+	killChan := make(chan struct{})
+	defer close(killChan)
 
-	go func() {
-		for {
-			select {
-			case <-time.After(5 * time.Millisecond):
-				err := proc.Signal(unix.SIGUSR1)
-				if err != nil {
-					t.Fatalf("Signal failed: %v", err)
-				}
-			case <-killchan:
-				return
-			}
-		}
-	}()
+	doneChan := make(chan struct{})
+	errChan := make(chan error, 1000)
 
 	go func() {
+		after := time.After(3 * time.Second)
+		i := 0
 		for {
 			select {
-			case <-time.After(11 * time.Millisecond):
-				err := w.poller.wake()
-				if err != nil {
-					t.Fatalf("Wake failed: %v", err)
-				}
-			case <-killchan:
+			case <-killChan:
+				close(doneChan)
 				return
-			}
-		}
-	}()
+			case <-after:
+				for ; i > 0; i-- {
+					testFile := fmt.Sprintf("%s%d", testFilePrefix, i)
+					err = os.Remove(testFile)
+					if err != nil {
+						errChan <- fmt.Errorf("Remove failed: %v", err)
+					}
+
+					time.Sleep(time.Millisecond)
```

Comment:
This stress test is pretty lengthy, so it might be nice to add a few more comments for future maintainers.